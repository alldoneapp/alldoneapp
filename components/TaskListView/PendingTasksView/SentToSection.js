import React from 'react'
import { StyleSheet, View } from 'react-native'
import { useSelector } from 'react-redux'

import ParentTaskContainer from '../ParentTaskContainer'
import WorkflowHeader from '../Header/WorkflowHeader'
import { translate } from '../../../i18n/TranslationService'

export default function SendToSection({ taskList, subtaskByTask, projectId, currentStepId, currentStep, assignee }) {
    const usersInProject = useSelector(state => state.projectUsers[projectId])

    const reviewer = usersInProject.find(user => user.uid === currentStep.reviewerUid)

    return (
        <View style={localStyles.container}>
            <WorkflowHeader
                reviewer={reviewer}
                assignee={assignee}
                projectId={projectId}
                currentStepId={currentStepId}
                workflowDirectionText={translate('sent to')}
            />
            {taskList.map(task => {
                const subtaskList = subtaskByTask[task.id] ? subtaskByTask[task.id] : []
                return (
                    <ParentTaskContainer
                        key={task.id}
                        task={task}
                        projectId={projectId}
                        subtaskList={subtaskList ? subtaskList : []}
                        isPending={true}
                    />
                )
            })}
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        flexDirection: 'column',
        marginLeft: 2,
    },
})
