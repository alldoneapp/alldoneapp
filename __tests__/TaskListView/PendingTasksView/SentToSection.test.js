/**
 * @jest-environment jsdom
 */

import React from 'react'
import renderer from 'react-test-renderer'
import SentToSection from '../../../components/TaskListView/PendingTasksView/SentToSection'


jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn().mockImplementation(fnc => {
        return fnc({
            loggedUserProjects: [{ id: '0' }],
            projectsUsers: {'0':[{ uid: 0, workflow: [] }]}
        })
    }),
    useDispatch: jest.fn()
}));

describe('SentToSection component', () => {
    it('should render correctly', () => {
        var element = React.createElement(SentToSection, {
            projectId: '0',
            tasks: [null, [{ userId: 0, stepHistory: [], userIds: [{}], recurrence: { type: 'never' } }]]
        })
        var json = renderer.create(element).toJSON()
        expect(json).toMatchSnapshot()
    })
})
