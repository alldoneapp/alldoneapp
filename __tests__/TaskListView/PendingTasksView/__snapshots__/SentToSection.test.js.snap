// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SentToSection component should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "height": 40,
        "paddingBottom": 2,
        "paddingLeft": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#8A94A6",
              "borderRadius": 100,
              "height": 24,
              "justifyContent": "center",
              "width": 24,
            },
            undefined,
            Object {
              "height": 24,
              "width": 24,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 100,
                "height": 20,
                "width": 20,
              },
              Object {
                "height": 20,
                "width": 20,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#8A94A6",
            "height": 2,
            "width": 4,
          }
        }
      />
      <View
        style={
          Object {
            "marginLeft": -2,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#8A94A6",
                "fontFamily": "alldone",
                "fontSize": 20,
                "opacity": 1,
              }
            }
          >
            
          </Text>
        </View>
      </View>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "marginLeft": 8,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#04142F",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 12,
                  "letterSpacing": 0.48,
                  "lineHeight": 20,
                },
                Object {
                  "color": "#8A94A6",
                },
              ]
            }
          >
            sent to Loading... • Loading...
          </Text>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
  <View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "space-between",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#EAFFF8",
              "borderRadius": 4,
              "flexDirection": "row",
              "height": "100%",
              "paddingLeft": 12,
              "width": "50%",
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#00C282",
                  "fontFamily": "alldone",
                  "fontSize": 18,
                  "opacity": 1,
                }
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "marginLeft": 4,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#04142F",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "letterSpacing": 0.16,
                    "lineHeight": 22,
                  },
                  Object {
                    "color": "#00C282",
                  },
                ]
              }
            >
              Details
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFF6EB",
              "borderRadius": 4,
              "flexDirection": "row",
              "height": "100%",
              "paddingRight": 12,
              "width": "50%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginLeft": "auto",
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#FFAE47",
                    "fontFamily": "alldone",
                    "fontSize": 18,
                    "opacity": 1,
                  }
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "marginLeft": 4,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#04142F",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "letterSpacing": 0.16,
                      "lineHeight": 22,
                    },
                    Object {
                      "color": "#FFAE47",
                    },
                  ]
                }
              >
                Reminder
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onLayout={[Function]}
        style={
          Object {
            "overflow": "visible",
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "flexDirection": "row",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -10000,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "width": 150,
              }
            }
          />
          <View
            onLayout={[Function]}
          />
        </View>
        <View
          style={
            Object {
              "bottom": 0,
              "flexDirection": "row-reverse",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -10000,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "width": 150,
              }
            }
          />
          <View
            onLayout={[Function]}
          />
        </View>
        <View
          collapsable={false}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          pointerEvents="auto"
          style={
            Object {
              "transform": Array [
                Object {
                  "translateX": -0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "justifyContent": "center",
                "marginLeft": -16,
                "marginRight": -16,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 4,
                }
              }
            >
              <View
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "rgba(241, 243, 244, 1)",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginHorizontal": 16,
                    "paddingHorizontal": 4,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    accessible={false}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "white",
                            "borderColor": "#8A94A6",
                            "borderRadius": 4,
                            "borderWidth": 1,
                            "height": 24,
                            "width": 24,
                          },
                          null,
                          Object {
                            "height": 20,
                            "width": 20,
                          },
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "flexGrow": 1,
                          "paddingLeft": 12,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      onLayout={[Function]}
                      onPress={[Function]}
                      style={
                        Array [
                          Array [
                            Object {
                              "color": "#04142F",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 14,
                              "letterSpacing": 0.32,
                              "lineHeight": 22,
                            },
                            Object {
                              "alignItems": "center",
                              "display": "flex",
                              "paddingBottom": 8,
                              "paddingTop": 8,
                            },
                            undefined,
                            Object {
                              "paddingBottom": 9,
                              "paddingTop": 9,
                            },
                            undefined,
                          ],
                          Object {
                            "flexDirection": "row",
                            "textAlignVertical": "center",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                            },
                            undefined,
                          ]
                        }
                      />
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingBottom": 8,
                        "paddingTop": 8,
                      },
                      Object {
                        "paddingBottom": 8,
                        "paddingTop": 8,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
